GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -44,17 +46,21 @@ public void ResetCache()
                 return null;
             }
 
-            // Callers setting branch name has the responsibility to ensure that not all are needed
-            if (string.IsNullOrWhiteSpace(branchName) && !string.IsNullOrWhiteSpace(_branchName))
+            lock (_lock)
             {
-                DebugHelpers.Fail($"Unexpectedly call for all branches after cache filled with specific branch {_branchName}");
-                ResetCache();
-            }
+                // Callers setting branch name has the responsibility to ensure that not all are needed
+                if (string.IsNullOrWhiteSpace(branchName) && !string.IsNullOrWhiteSpace(_branchName))
+                {
+                    Trace.WriteLine($"Unexpectedly call for all branches after cache filled with specific branch {_branchName}");
+                    ResetCache();
+                }
 
-            // Use Lazy<> to synchronize callers
-            _lazyData ??= new(() => GetData(null, branchName));
-            _branchName = branchName;
-            return _lazyData.Value;
+                // Use Lazy<> to synchronize callers
+                _branchName = branchName;
+                Trace.WriteLine("_lazyData:" + _lazyData + "/bn:" + _branchName);
```

Comment:
Use string interpolation and do not abbreviate?
```suggestion
                Trace.WriteLine($"{nameof(_lazyData)}: {_lazyData}, {nameof(_branchName)}: {_branchName}");
```

Reply:
Do you want to add this printout?

Reply:
No that's why I removed it when I was aware of this mistake.